api/lib/opentelemetry/internal/proxy_tracer_provider.rb

```diff
@@ -37,21 +37,22 @@ def delegate=(provider)
 
         @mutex.synchronize do
           @delegate = provider
-          @registry.each { |key, tracer| tracer.delegate = provider.tracer(key.name, key.version) }
+          @registry.each { |key, tracer| tracer.delegate = provider.tracer(key.name, key.version, key.schema_url) }
         end
       end
 
       # Returns a {Tracer} instance.
       #
       # @param [optional String] name Instrumentation package name
       # @param [optional String] version Instrumentation package version
+      # @param [optional String] schema_url Schema URL to be recorded with traces
       #
       # @return [Tracer]
-      def tracer(name = nil, version = nil)
+      def tracer(name = nil, version = nil, schema_url = nil)
```

Comment:
Can we make `schema_url` a named parameter rather than positional? At this point, I wish we'd made `version` an optional named parameter 😞

Reply:
Yes - and given the upcoming [instrumentation attributes stuff](https://github.com/open-telemetry/opentelemetry-ruby/issues/1262) I think that's a wise idea; will change.

> At this point, I wish we'd made version an optional named parameter 😞

That would be nice, yes. We could do something like this, I think:

```ruby
      def tracer(_name = nil, _version = nil, name: nil, version: nil, schema_url: nil)
        ...
        name ||= _name
        version ||= _version
```

...but that might be even more distasteful. Regardless, I'll make `schema_url` a named param.

Reply:
Update: I went through and changed this to kwargs just about everywhere; except for the internal _SDK_ tracer initialization. We could do it there as well, but I was lazy and didn't feel like updating anything else since that's `@api private` anyways.

(But if you feel strongly, I can go further).